crossmap/1.0/convert_maf_coords.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# This lcr-script is adopted from Bruno's convert_maf_coords.sh script available at Morin Lab Scripts
+# github repo. Some modifications are adopted to allow its implementation as a part of vcf2maf lcr-module.
+
+#   Usage:
+#       convert_maf_coords.sh input.maf hg38ToHg19.over.chain.gz output.maf [crossmap|liftover]
+#
+#   The default mode uses CrossMap.py to convert between genomic coordinate systems, because it
+#   tends to drop fewer variants than liftOver. 
+#
+#   The script assumes that there are no pipe (|) characters in your MAF file. If this is not
+#   the case, you can specify your own MAF column separator using the MAFCOLSEP variable.
+#
+#   By default, the script expects CrossMap.py or liftOver to be in your PATH. If that's not the
+#   case, you can specify the program path using the CROSSMAP or LIFTOVER environment variables.
+#
+# ----------------------------------------------------------------------------------------------- #
+
+
+set -euf -o pipefail
+
+# Parameters
+SCRIPT="$(basename $0)"
+INPUT_MAF="$1"
+INPUT_BED="$(mktemp /tmp/${SCRIPT}.input.XXXXXX)"
+CHAIN_FILE="$2"
+OUTPUT_BED="$(mktemp /tmp/${SCRIPT}.output.XXXXXX)"
+OUTPUT_MAF="$3"
+OUTPUT_UNMAPPED="${OUTPUT_MAF%.*}.unmapped.bed"
+MODE="${4:-crossmap}"
+MAFCOLSEP="${MAFCOLSEP:-|}"
+DEBUG="${DEBUG:-0}"
+
+# For colour
+RED='\033[0;31m'
+NC='\033[0m'
+
+DATE=$(date --rfc-3339 seconds | cut -c1-19)
+echo "[${DATE}] INFO: Checking whether there are \"${MAFCOLSEP}\" characters in the input" \
+	"MAF file."
+if fgrep -q \'${MAFCOLSEP}\' "${INPUT_MAF}"; then
+	echo -e "${RED}[${DATE}] ERROR: ${INPUT_MAF} contains \"${MAFCOLSEP}\" characters, which" \
+		"are used to separate MAF columns in the intermediate BED files. Please set the" \
+		"MAFCOLSEP environment variable to change this.${NC}"; exit 1;
+fi
+
+DATE=$(date --rfc-3339 seconds | cut -c1-19)
+echo "[${DATE}] INFO: Creating temporary BED file (${INPUT_BED})" \
+	"from input MAF file."
+awk 'BEGIN {FS=OFS="\t"} \
+		{chrom = $5; start = $6; end = $7} \
+		$10 == "SNP" {start = start - 1; end = end} \
+		$10 == "DEL" {start = start - 1; end = end} \
+		$10 == "INS" {start = start; end = end - 1} \
+		$0 !~ /^(#|Hugo_Symbol)/ {row=gensub(/\t/, "'"${MAFCOLSEP}"'", "g", $0); print chrom, start, end, row}' \
+		"${INPUT_MAF}" | \
+	sed 's/ /_____/g' > "${INPUT_BED}"
+
+if [[ "${MODE}" == "crossmap" ]]; then
+	CROSSMAP="${CROSSMAP:-CrossMap.py}"
+	command -v "${CROSSMAP}" >/dev/null 2>&1 || { \
+		echo -e "${RED}[${DATE}] ERROR: ${CROSSMAP} is not in the PATH or the file doesn't" \
+			"exist.${NC}"; exit 1; }
+	DATE=$(date --rfc-3339 seconds | cut -c1-19)
+	echo "[${DATE}] INFO: Converting genomic coordinates using ${CROSSMAP}."
+	"${CROSSMAP}" bed "${CHAIN_FILE}" "${INPUT_BED}" "${OUTPUT_BED}"
+	mv -f "${OUTPUT_BED}.unmap" "${OUTPUT_UNMAPPED}"
+else
+	DATE=$(date --rfc-3339 seconds | cut -c1-19)
+	LIFTOVER="${LIFTOVER:-liftOver}"
```

Comment:
This seems to assume the user has liftOver in their environment. Should it not be added to the environment for this script?

Reply:
I will add it to the environment.

Reply:
It's only using either liftover OR crossmap, right? IMO there should be a separate yaml file for liftover if folks want to use it over crossmap.

Reply:
yes, it is using either of these tools. I was considering removing liftover from this script, since we already have module for it. I think adding liftover to the environment will simplify the use of the script.

Reply:
IMHO there is little benefit to splitting this up into two environments as both are quite lightweight. I agree with having them combined into one environment.